web/impact/impact/tests/test_user_detail_view.py

```diff
@@ -628,6 +630,36 @@ def test_patch_user_with_no_profile_not_allowed_on_missing_fields(self):
             assert MISSING_PROFILE_ERROR.format(user.id) in response.data
             assert MISSING_SUBJECT_ERROR in response.data
 
+    def test_get_user_latest_active_program_location(self):
+        program = ProgramFactory(
+            program_status=ACTIVE_PROGRAM_STATUS,
+            program_family=ProgramFamilyFactory(),
+        )
+        context = UserContext(user_type=EXPERT_USER_TYPE, program=program)
```

Comment:
Rather than adding the program role grant in the context, let's do it here:
```
role_grant = ProgramRoleGrantFactory(program_role__program_status=ACTIVE_PROGRAM_STATUS)
```
Then we can get the user_program from the role_grant:
```
...
user_program=role_grant.program_role.program
...
``` 
(line 643)